src/connector/src/common.rs

```diff
@@ -371,7 +371,15 @@ impl NatsCommon {
         if let (Some(v_user), Some(v_password)) = (self.user.as_ref(), self.password.as_ref()) {
             connect_options = connect_options.user_and_password(v_user.into(), v_password.into());
         }
-        let client = connect_options.connect(self.server_url.clone()).await?;
+        let servers = self.server_url.split(',').collect::<Vec<&str>>();
+        let client = connect_options
+            .connect(
+                servers
+                    .iter()
+                    .map(|url| url.parse())
+                    .collect::<Result<Vec<async_nats::ServerAddr>, _>>()?,
```

Comment:
we can discard the wrong addr and keep the valid ones. IDK if there is internal communication among the brokers. For kafka, we just need one valid addr and the broker will redirect to the lead broker. 

& please refine the error message, it may be confusing get the parsing error.

Reply:
1. I add a `map_err`, but if url error, it will throw error in other place like this : `ERROR:  QueryError: internal error: Rpc error: gRPC error (Internal error): failed to send barrier to actor 40: SendError { .. }`. Map error can only catch build client  error.  I try to use some rust url package to catch url error. But seems always regard a correct url as false. 
2. I suggest to return error rather than keep the right and discard wrong addrs. For Nats, at least when I try it in test, subscribe one address or two address will make difference in the end. I didn't find docs about the inner logic of their multiple address. So I prefer return error.